Update README + fixes startup (#76)
DAGWorks-Inc/burr: README.md

# Burr

> Burr provides tooling to build applications that make decisions (chatbots, agents, simulations, etc...) from simple python building blocks.

Reviewer: ```suggestion
Burr is a lightweight framework to build applications that make decisions based on state (chatbots, agents, simulations, etc...) from simple python building blocks.
```

> Burr includes a UI that can track/monitor those decisions in real time.

Reviewer: ```suggestion
Burr includes a UI that can track, monitor, debug, and annotate those decisions in real time.
```

Link to [documentation](https://burr.dagworks.io/). Quick video intro [here](https://www.loom.com/share/8a92474bb7574d6eb4cd25c21913adf2).

## 🏃Quick start

Install from `pypi`:

```bash
pip install "burr[start]"
```

Then run the server:

```bash
burr
```

> This will open up a demo -- to chat it requires the `OPENAI_API_KEY` environment variable to be set,

Reviewer: ```suggestion
This will open up the burr tracking UI with demo data and an example built in application -- to use the demo chat application, it requires the `OPENAI_API_KEY` environment variable to be set,
```

> but you can still see how it works if you don't have one.

Reviewer: ```suggestion
but you can still see how the telemetry UI works if that is not set.
```

## 🔩 How does Burr work?

With Burr you express your application as a state machine (i.e. a graph/flowchart).
You can (and should!) use it for anything where managing state can be hard. Hint: managing state is always hard!
This is true across a wide array of contexts, from building RAG applications to power a chatbot, to running ML parameter tuning/evaluation workflows,
to conducting a complex forecasting simulation.

Burr includes:

1. A (dependency-free) low abstraction python library that enables you to build and manage state machines with simple python functions

> 2. A UI you can use view execution telemetry for introspection and debugging

Reviewer: ```suggestion
2. A UI you can use view execution state & telemetry for introspection, debugging, and annotating
```

> 3. A set of integrations to make it easier to persist state, connect to telemetry, and integrate with other systems

Reviewer: ```suggestion
3. A pluggable interface to make it easier to persist state, connect to telemetry, and integrate with other systems
```
